apps/builder/app/routes/s/css.ts

```diff
@@ -5,46 +5,98 @@ import env from "~/env/env.public.server";
 import { getBuildParams } from "~/shared/router-utils";
 import { sentryException } from "~/shared/sentry";
 import { createContext } from "~/shared/context.server";
-import { loadCanvasData } from "~/shared/db";
+import { loadCanvasData, loadProductionCanvasData } from "~/shared/db";
+import type { Tree } from "@webstudio-is/project-build";
 
 export const loader = async ({ request }: ActionArgs) => {
   try {
     const buildParams = getBuildParams(request);
-    const context = await createContext(request);
+
+    const buildEnv = buildParams?.mode === "published" ? "prod" : "dev";
+
+    const context = await createContext(request, buildEnv);
 
     if (buildParams === undefined) {
       throw json("Required project info", { status: 400 });
     }
 
     const project = await db.project.loadByParams(buildParams, context);
+
     if (project === null) {
       throw json("Project not found", { status: 404 });
     }
 
-    const canvasData = await loadCanvasData(
-      {
-        project,
-        env: buildParams.mode === "published" ? "prod" : "dev",
-        pageIdOrPath:
-          "pageId" in buildParams ? buildParams.pageId : buildParams.pagePath,
-      },
+    if (buildEnv === "dev") {
+      const canvasData = await loadCanvasData(
+        {
+          project,
+          env: buildParams.mode === "published" ? "prod" : "dev",
+          pageIdOrPath:
+            "pageId" in buildParams ? buildParams.pageId : buildParams.pagePath,
+        },
+        context
+      );
+
+      if (canvasData === undefined) {
+        throw json("Page not found", { status: 404 });
+      }
+
+      const cssText = generateCssText({
+        assets: canvasData.assets,
+        breakpoints: canvasData.build?.breakpoints,
+        styles: canvasData.build?.styles,
+        styleSourceSelections: canvasData.tree?.styleSourceSelections,
+      });
+
+      return new Response(cssText, {
+        headers: {
+          "Content-Type": "text/css",
+          // We have no way with Remix links to know if the CSS has changed (no ?cache-breaker in url)
+          // we can add Last-Modified and change on Cache-Control: no-cache but this is used only for localhost publish
+          // And can be fully omitted for the Designer Canvas. (_Not an issue on SaaS as we know data at the build time_)
```

Comment:
Isn't it more correct "Not an issue on a published site..."?
or would it be a problem for an OSS user who deploys to vercel for example?